agent/structs/structs.go

```diff
@@ -1333,9 +1333,8 @@ func (s *NodeService) Validate() error {
 					"services"))
 		}
 
-		if s.Port == 0 && s.SocketPath == "" {
```

Comment:
This makes me think that we should require one of port or socket path everywhere.

Reply:
Added. Let me know if it is not the case for gateway types - in which case I will remove the sockpath validation below.

Reply:
Removed for gateway types per our discussion.